File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/7 - 9 and 10 March 2019/06. High Jump.py
aim = int(input())
start = aim - 30
bad_count = 0
total_jumps = 0

while not(start > aim):
    if bad_count == 3:
        break
    jump = int(input())
    total_jumps += 1
    if jump > start:
        start += 5
        bad_count = 0
    else:
        bad_count += 1

if bad_count == 3:
    print(f"Tihomir failed at {start}cm after {total_jumps} jumps.")
else:
    print(f"Tihomir succeeded, he jumped over {start - 5}cm after {total_jumps} jumps.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/03. Online Exam - 9 and 10 March 2019/06. High Jump.js
function highJump(input) {
    // Mask
    let aim = Number(input.shift());
    let start = aim - 30;
    let badCount = 0;
    let totalJumps = 0;
    
    while (!(start > aim)) {
        if (badCount == 3) {break;}
        let jump = Number(input.shift());
        totalJumps += 1;
        if (jump > start) {
            start += 5;
            badCount = 0;
        } else {
            badCount += 1;
        }
    }

    if (badCount == 3) {
        console.log(`Tihomir failed at ${start}cm after ${totalJumps} jumps.`);
    } else {
        console.log(`Tihomir succeeded, he jumped over ${start - 5}cm after ${totalJumps} jumps.`)
    }
}

highJump([231, 205, 212, 213, 228, 229, 230, 235]);  // Should return:
// Tihomir succeeded, he jumped over 231cm after 7 jumps.

highJump([250, 225, 224, 225, 228, 231, 235, 234, 235]);  // Should return:
// Tihomir failed at 235cm after 8 jumps.